sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -139,6 +139,7 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
   object JoinSelection extends Strategy
     with PredicateHelper
     with JoinSelectionHelper {
+    private val hintErrorHandler = conf.hintErrorHandler
```

Comment:
can we check this in `ResolveJoinStrategyHints`?

Reply:
I thought about it but seems we need to do this check in physical side.

We don't konw the join strategy at analysis or optimizer (`EliminateResolvedHint`). For example, if user specify `boradcast`, we need to check if it's a equijoin first then check the build side, since this join could be bhj or bnlj.

Reply:
maybe we can move `ExtractEquiJoinKeys` to catalyst?

Reply:
It's a point. How about make a new rule at Optimizer `Finish Analysis` batch to check if join is illegal. Move to Optimizer is to avoid effect like `EliminateOuterJoin`.

Reply:
`EliminateOuterJoin` is a good point, so we can only know if the hint is valid or not at the planner.

It's a bit unfortunate to see the hint code being widespread, @maryannxue do you have any thoughts?

Reply:
How about move `EliminateResolvedHint` from optimizer to physical side ? then we can add the check in one rule.